src/item_factory.cpp

```diff
@@ -884,6 +901,33 @@ void Item_factory::load_gunmod(JsonObject &jo)
     load_basic_info( jo, new_item_template );
 }
 
+void Item_factory::load( islot_magazine &slot, JsonObject &jo )
+{
+    slot.type = jo.get_string( "ammo_type" );
```

Comment:
Once the implementation is stable I'll add some more documentation. It's likely that I'll follow @BevapDins suggestion in #14149 and add a call to `load_slot_optional` but again this can wait for another PR.